src/main/java/com/fasterxml/jackson/databind/type/SimpleType.java

```diff
@@ -53,7 +53,7 @@ protected SimpleType(Class<?> cls, TypeBindings bindings,
             Object valueHandler, Object typeHandler, boolean asStatic)
     {
         super(cls, bindings, superClass, superInts,
-                0, valueHandler, typeHandler, asStatic);
+                SimpleType.class.hashCode(), valueHandler, typeHandler, asStatic);
```

Comment:
Does this make any real difference? (I guess it won't hurt)

Reply:
Ah. So make it less likely we get `0`. For that purpose any old constant would do tho, 1, 31 whatever.